openid_connect.js

```diff
@@ -7,7 +7,43 @@ var newSession = false; // Used by oidcAuth() and validateIdToken()
 
 export default {auth, codeExchange, validateIdToken, logout};
 
+function sleep(ms) {
+    return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+// If the ID token has not been synced yet, poll the variable every 100ms until
+// get a value or after a timeout.
+async function waitForSessionSync(r, timeout, checkInterval) {
+    let timePass = 0;
+    do {
+        await sleep(checkInterval);
+        timePass += checkInterval;
+    } while (!r.variables.session_jwt && timePass < timeout);
+
+    if (r.variables.session_jwt) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
 function auth(r) {
+    // If a cookie was sent but the ID token is not in the key-value database, wait for the token to be in sync.
+    if (r.variables.cookie_auth_token && !r.variables.session_jwt && !newSession && r.variables.zone_sync_leeway > 0) {
+        waitForSessionSync(r, r.variables.zone_sync_leeway, 100)
+        .then(result => {
+            if (result) {
+                delete r.headersOut["WWW-Authenticate"]; // Remove evidence of original failed auth_jwt
+                r.internalRedirect(r.variables.uri + r.variables.is_args + (r.variables.args || ''));
+            } else {
+                newSession = true;
```

Comment:
That changes existing logic in case of refresh token presence. I'd suggest to add new argument to `auth` or even factor out `startNewSession` function, instead.

Reply:
Thank you, added "afterSyncCheck" argument to the function "auth()"